src/Umbraco.Cms.Api.Management/Factories/TemplatePresentationFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Umbraco.Cms.Api.Management.ViewModels.Template;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Api.Management.Factories;
+
+public class TemplatePresentationFactory : ITemplatePresentationFactory
+{
+    private readonly ITemplateService _templateService;
+    private readonly IUmbracoMapper _mapper;
+
+    public TemplatePresentationFactory(
+        ITemplateService templateService,
+        IUmbracoMapper mapper)
+    {
+        _templateService = templateService;
+        _mapper = mapper;
+    }
+
+    public async Task<TemplateResponseModel> CreateTemplateResponseModelAsync(ITemplate template)
+    {
+        TemplateResponseModel responseModel = _mapper.Map<TemplateResponseModel>(template)!;
```

Comment:
Unsure if we actually need to use the mapper here, or if we should just have the "mapping" logic directly in the method. 🤔 Presumably, this should be the only place where we are creating a `TemplateResponseModel` anyways.

Reply:
Yeah that's a very good point 😄 I'll update it